daemon/list.go

```diff
@@ -234,6 +234,15 @@ func includeContainerInList(container *Container, ctx *listContext) iterationAct
 		return excludeContainer
 	}
 
+	// Do not include container if the isolation mode doesn't match
+	i := strings.ToLower(string(container.hostConfig.Isolation))
```

Comment:
should we make this a platform function? it could always be `true` for unix and check the value for windows.